### PR TITLE
Migrate code to support Lab3 (breaks lab2 compat)

### LINKED ISSUE #1: Compatibility with Lab3
Hi guys, 

Lab 3 was released early this year and brings some changes on how extensions are built and deployed. The current extension is thus incompatible. How can we proceed to update this extension?

The Jupyter team has provided a guide and even some tools to ease the transition: https://jupyterlab.readthedocs.io/en/latest/extension/extension_migration.html

Diogo

### PR DESCRIPTION
**Pull request by didithilmy**
Migrate codebase to support JL3 according to this guide: https://jupyterlab.readthedocs.io/en/latest/extension/extension_migration.html. Breaks compatibility to JL2.

Some quick fixes are used:
1. The use of `webpack.config.js` with `fullySpecified: false` option is somehow required, or else the build fails. Might be a bug on Babel/Webpack (https://github.com/webpack/webpack/issues/11467)
2. The use of `resolutions` on `package.json` is required due to dependency conflict between the new JL3 `@types/react` dependency and `react-window`.
Fixes #1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ for the frontend extension.
 
 ## Requirements
 
-* JupyterLab >= 2.0
+* JupyterLab >= 3.0
 
 ## Install
```
Comment (diocas) [inline README.md:18]: The installation instructions changed. We no longer need to build jupyterlab

File: jupyter-config/rucio_jupyterlab.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
-  "NotebookApp": {
-    "nbserver_extensions": {
+  "ServerApp": {
+    "jpserver_extensions": {
```
Comment (diocas) [inline jupyter-config/rucio_jupyterlab.json:3]: It's normal that the script changes from a notebook server extension to a jupyter server extension. But I don't see in your changes the upgrade. Did you check if it works as is? But, even if that is the case, can we have the migration of the server as well? https://jupyter-server.readthedocs.io/en/latest/developers/extensions.html#migrating-an-extension-to-use-jupyter-server

**Review by diocas — CHANGES_REQUESTED** [review]
I took the liberty to take a look, even though this is now Rucio's turf :) Hope you don't mind.
Looks good, just a few comments.

Since the big change was in the way Lab packages and deploys the extensions, if you manage to open it, there should be no problems. But I also mentioned that it would be good to migrate to jupyter server extension, so I hope this doesn't bring any other issue.

File: jupyter-config/rucio_jupyterlab.json (hunk 2 above)
Comment (didithilmy) [reply jupyter-config/rucio_jupyterlab.json]: I've pushed the changes, please check :)

**Comment by didithilmy** [comment]
Ah, one problem: If I run `python -m jupyterlab.browser_check`, it throws an error with the following stack trace:
```
Parsed an error from text content: undefined Warning: React does not recognize the `%s` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `%s` instead. If you accidentally passed it from a parent component, remove it from the DOM element.%s

/Users/mhilmy/Documents/GitHub/jupyterlab-extension/venv/share/jupyter/lab/browser_test/chrome-test.js:55
  throw up;
  ^
true
(Use `node --trace-uncaught ...` to show where the exception was thrown)
[C 2021-02-25 17:18:03.685 ServerApp] Caught exception during the test:
Caught exception during the test:
[E 2021-02-25 17:18:03.685 ServerApp] ['node', 'chrome-test.js', 'file:///Users/mhilmy/Library/Jupyter/runtime/jpserver-2836-open.html'] exited with 1
['node', 'chrome-test.js', 'file:///Users/mhilmy/Library/Jupyter/runtime/jpserver-2836-open.html'] exited with 1
[I 2021-02-25 17:18:03.685 ServerApp] Test Complete
[C 2021-02-25 17:18:03.685 ServerApp] Exiting with 1 due to errors
```

I'm not sure why, maybe you have an experience with this? @diocas

**Comment by didithilmy** [comment]
> Ah, one problem: If I run `python -m jupyterlab.browser_check`, it throws an error with the following stack trace:
> 
> ```
> Parsed an error from text content: undefined Warning: React does not recognize the `%s` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `%s` instead. If you accidentally passed it from a parent component, remove it from the DOM element.%s
> 
> /Users/mhilmy/Documents/GitHub/jupyterlab-extension/venv/share/jupyter/lab/browser_test/chrome-test.js:55
>   throw up;
>   ^
> true
> (Use `node --trace-uncaught ...` to show where the exception was thrown)
> [C 2021-02-25 17:18:03.685 ServerApp] Caught exception during the test:
> Caught exception during the test:
> [E 2021-02-25 17:18:03.685 ServerApp] ['node', 'chrome-test.js', 'file:///Users/mhilmy/Library/Jupyter/runtime/jpserver-2836-open.html'] exited with 1
> ['node', 'chrome-test.js', 'file:///Users/mhilmy/Library/Jupyter/runtime/jpserver-2836-open.html'] exited with 1
> [I 2021-02-25 17:18:03.685 ServerApp] Test Complete
> [C 2021-02-25 17:18:03.685 ServerApp] Exiting with 1 due to errors
> ```
> 
> I'm not sure why, maybe you have an experience with this? @diocas

Okay never mind, after I fixed the config files to match Lab3 migration guide the problem went away.

**Review by diocas — APPROVED** [review]
Looks good to me!

**Comment by bari12** [comment]
👍

**Comment by didithilmy** [comment]
Great! I'll merge this to master. The Docker image should update to the latest Lab version as well.

**Pull request merged**